core/src/main/java/org/mskcc/cbio/portal/scripts/ImportProfileData.java

```diff
@@ -87,6 +87,13 @@ public void run() {
             } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.FUSION) {
                 ImportFusionData importer = new ImportFusionData(dataFile, geneticProfile.getGeneticProfileId(), genePanel);
                 importer.importData();
+            } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.GENERIC_ASSAY) {
+
+                // add all missing `genetic_entities` for this assay to the database
+                ImportGenericAssayEntity.importData(dataFile);
+                
+                ImportTabDelimData genericAssayProfileImporter = new ImportTabDelimData(dataFile, geneticProfile.getTargetLine(), geneticProfile.getGeneticProfileId(), genePanel);
```

Comment:
Not suggesting you do it now, but did you consider importing generic assay data in same process as importing genetic assay entity?

Reply:
Yes, but we decided to divide responsibility of for different tables/data types to different importer scripts. Its is a matter of taste I guess. I like that looking from the outside it is clear that the Entities are loaded, followed by import of the profile data. Creating a dedicated Importer class that does both for generic assay data is possible in the future and can be discussed.

Reply:
@dippindots is working on implementing proper support for generic_assay_support.  As part of this effort, he will refactor this code (after PR is merged) into a general solution.  We've had a few conversations at MSK about the approach to take, but haven't discussed in detail how the import process would change.  If you would like to be included in this discussion, let us know (just assumed you had other things to do).